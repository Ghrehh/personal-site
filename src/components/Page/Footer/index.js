// @flow
import * as React from 'react';

import SectionHeading from 'components/text/SectionHeading';
import Button from 'components/Button';

import styles from './styles.module.css';

const Footer = () => (
  <React.Fragment>
    <div className={styles.slantedBackground} />

    <footer className={styles.footer}>
      <div className={styles.footerInner}>
        <div className={styles.textContainer}>
          <SectionHeading className={styles.heading}>CONTACT</SectionHeading>
          <h2 className={styles.subHeading}>Feel free to get in touch.</h2>
        </div>

        <div>
          <Button className={styles.twitterButton}>Twitter</Button>
          <Button className={styles.emailButton}>Email</Button>
          <Button className={styles.linkedInButton}>LinkedIn</Button>
        </div>
      </div>
    </footer>
  </React.Fragment>
);

export default Footer;
