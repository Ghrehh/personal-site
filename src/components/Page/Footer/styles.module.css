.slantedBackground {
  position: relative;
}

.slantedBackground:after {
  content: "";
  position: absolute;
  background-color: var(--light-grey);
  clip-path: polygon(0 40%, 100% 0, 100% 100%, 0% 100%);
  height: 170px;
  width: 100%;
  z-index: -1;
  bottom: 0;
}

.footer {
  padding: 53px;
  background-color: var(--light-grey);
}

.footerInner {
  max-width: 850px;
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.textContainer {
  margin-right: 10px;
}

.heading {
  margin-bottom: 1px;
}

.subHeading {
  font-size: 24px;
  color: var(--green);
}

.twitterButton {
  width: 150px;
  text-align: center;
  margin-right: 10px;
  background-color: var(--blue);
}

.emailButton {
  width: 150px;
  text-align: center;
  background-color: var(--red);
}

@media (max-width: 708px) {
  .footerInner {
    justify-content: center;
  }

  .textContainer {
    margin-right: 20px;
  }

  .twitterButton {
    display: block;
    margin-right: 0;
    margin-bottom: 10px;
  }
}

@media (max-width: 450px) {
  .twitterButton, .emailButton {
    display: block;
    margin-right: auto;
    margin-left: auto;
  }

  .footerInner {
    display: block;
  }

  .textContainer {
    margin-right: 0;
    margin-bottom: 25px;
    text-align: center;
  }
}
