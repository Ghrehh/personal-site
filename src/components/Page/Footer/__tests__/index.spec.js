import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Footer from '../';

describe('Footer', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(<Footer />);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
