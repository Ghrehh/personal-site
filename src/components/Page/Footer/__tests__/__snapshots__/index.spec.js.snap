// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the component correctly 1`] = `
<Unknown>
  <div
    className="slantedBackground"
  />
  <footer
    className="footer"
  >
    <div
      className="footerInner"
    >
      <div
        className="textContainer"
      >
        <SectionHeading
          className="heading"
        >
          CONTACT
        </SectionHeading>
        <h2
          className="subHeading"
        >
          Feel free to get in touch.
        </h2>
      </div>
      <div>
        <Button
          className="twitterButton"
        >
          Twitter
        </Button>
        <Button
          className="emailButton"
        >
          Email
        </Button>
      </div>
    </div>
  </footer>
</Unknown>
`;
